### PR TITLE
Feature/general polishing

### PR DESCRIPTION
**Pull request by monsagri**
Added test, removed unused types, updated readme.

Also changed the jest config slightly to allow for unit tests to be stored on the same level as the file they are testing. Happy to change that if it's not in convention.

Quite a lot of lines changed in this one, but a large amount of it comes from refactoring util methods to be pure functions for testing and moving them to util.ts

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2015 Stefan Tertan
+Copyright (c) 2015 MySense
```
Comment (ColdFire87) [inline LICENSE]:
```suggestion
Copyright (c) 2020 MySense
```

**Comment by MousyBusiness** [comment]
@monsagri I wouldn't worry so much about the lines, I occassionally get 20k lines from @ColdFire87

File: src/index.ts [hunk 2]
```diff
@@ -65,13 +63,11 @@ class ServerlessSSMPublish {
       'ssmPublish:upsertParams': this.hookWrapper.bind(this, this.updateParams.bind(this)),            // tslint:disable-line:no-unsafe-any
 
       // Serverless lifecycle events
-      'after:package:createDeploymentArtifacts': async () => { // check if this is the best place to call our plugin
+      'after:deploy:deploy': async () => { // check if this is the best place to call our plugin
```
Comment (ColdFire87) [inline src/index.ts]: could you give a short explanation why the change in events?

File: src/util.test.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+import { compareParams, evaluateEnabled, validateParams } from './util';
+
+const throwFunction = (message: string): void => { throw new Error(message); };
+const logFunction = (message: string): void => { console.log(message); }; // tslint:disable-line:no-console
+
+describe('evaluateEnabled should correctly validate if the plugin is enabled', () => {
+
+  test('It should throw if no ssmPublish settings are passed', () => {
+    const testEvalute = () => evaluateEnabled({ } as ServerlessInstance['service']['custom'], throwFunction); // tslint:disable-line:no-object-literal-type-assertion
+    expect(testEvalute).toThrow();
```
Comment (ColdFire87) [inline src/util.test.ts]: Typo

```suggestion
    const testEvaluate = () => evaluateEnabled({ } as ServerlessInstance['service']['custom'], throwFunction); // tslint:disable-line:no-object-literal-type-assertion
    expect(testEvaluate).toThrow();
```

File: src/util.test.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  });
+
+  test('It should throw if enabled is a string and not true or false', () => {
+    const testEvalute = () => evaluateEnabled({ ssmPublish: { enabled: 'bla'}} as ServerlessInstance['service']['custom'] , throwFunction); // tslint:disable-line:no-object-literal-type-assertion
+    expect(testEvalute).toThrow(); // tslint:disable-line:no-object-literal-type-assertion
```
Comment (ColdFire87) [inline src/util.test.ts]: Typo

```suggestion
    const testEvaluate = () => evaluateEnabled({ ssmPublish: { enabled: 'bla'}} as ServerlessInstance['service']['custom'] , throwFunction); // tslint:disable-line:no-object-literal-type-assertion
    expect(testEvaluate).toThrow(); // tslint:disable-line:no-object-literal-type-assertion
```

File: src/index.ts (hunk 2 above)
Comment (monsagri) [reply src/index.ts]: Sure.
Basically I think it should always have been run after deploy, but I was testing things by running `sls package`, which cuts out before getting to the deploy lifecycle event.

The main reason I'm leaning towards after deploy is because that way you won't need to rollback SSM variables that might have been created before a failed deployment (and that's the description you gave the repo on github).

File: src/util.test.ts (hunk 3 above)
Comment (monsagri) [reply src/util.test.ts]: Done

File: src/index.ts (hunk 2 above)
Comment (ColdFire87) [reply src/index.ts]: sounds like a wise choice 👍

File: src/util.test.ts (hunk 4 above)
Comment (monsagri) [reply src/util.test.ts]: Done

File: LICENSE (hunk 1 above)
Comment (monsagri) [reply LICENSE]: Done

File: LICENSE [hunk 5]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2015 Stefan Tertan
+Copyright (c) 2020 MySense
```
Comment (sorincezar) [inline LICENSE]: nice one
